src/ImageSharp/Formats/Gif/LzwDecoder.cs

```diff
@@ -93,10 +92,11 @@ public LzwDecoder(MemoryAllocator memoryAllocator, BufferedReadStream stream, in
         this.endCode = this.clearCode + 1;
         this.availableCode = this.clearCode + 2;
 
-        ref int suffixRef = ref MemoryMarshal.GetReference(this.suffix.GetSpan());
-        for (this.code = 0; this.code < this.clearCode; this.code++)
+        Span<int> suffix = this.suffixOwner.GetSpan();
+        int max = Math.Min(this.clearCode, suffix.Length);
+        for (this.code = 0; this.code < max; this.code++)
         {
-            Unsafe.Add(ref suffixRef, (uint)this.code) = (byte)this.code;
+            suffix[this.code] = (byte)this.code;
```

Comment:
This could run on a slice of `suffix`.

Reply:
Or if there is an upper limit to `Math.Min(this.clearCode, suffix.Length)`, it can be copied out from a statically initialized buffer.

Reply:
Clear code maxes out at 256. We can certainly slice here.